apis/tempo/v1alpha1/microservices_webhook.go

```diff
@@ -98,6 +100,24 @@ func (d *Defaulter) Default(ctx context.Context, obj runtime.Object) error {
 		r.Spec.SearchSpec.DefaultResultLimit = &defaultDefaultResultLimit
 	}
 
+	// Default replicas for distibutor
+	if r.Spec.Components.Distributor == nil {
```

Comment:
Could we change the `.Spec.Components.Distributor` (and friends) to a non-pointer struct?

Reply:
I would be good!! great suggestion